src/exceptionless.node.ts

```diff
@@ -116,12 +137,16 @@ process.on(BEFORE_EXIT, function (code:number) {
   }
 
   var client = ExceptionlessClient.default;
+  var config = client.config;
   var message = getExitCodeReason(code);
+
   if (message !== null) {
-    client.submitLog(BEFORE_EXIT, message, 'Error')
+    client.submitLog(EXIT, message, 'Error')
```

Comment:
As far as I can see, all plugins run synchronously (the run-signature gets a context object and no callback). If we ever run code here that depends on callbacks, this code would have to know the isApplicationExiting-state and run synchronously. 

But as long as submitEvent(E); queue.process(); work, this should work too.